### PR TITLE
Upgrade ASM to version 7.1

### PR DESCRIPTION
**Pull request by paplorinc**
Fixes: https://github.com/gradle/gradle/issues/9355
See: https://asm.ow2.io/versions.html

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/process/internal/worker/child/WorkerProcessClassPathProvider.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ private void remapClass(Class<?> classToMap, ZipOutputStream jar) throws IOExcep
                 inputStream.close();
             }
             ClassWriter classWriter = new ClassWriter(0);
-            ClassVisitor remappingVisitor = new RemappingClassAdapter(classWriter, remapper);
```
Comment (paplorinc) [inline subprojects/core/src/main/java/org/gradle/process/internal/worker/child/WorkerProcessClassPathProvider.java]: Removed in latest version (was deprecated for some time)

**Comment by paplorinc** [comment]
Hey @JLLeitschuh, thanks, you have to be more specific than that :)

**Comment by JLLeitschuh** [comment]
@paplorinc See this test:

https://github.com/gradle/gradle/blob/master/subprojects/distributions/src/integTest/groovy/org/gradle/DistributionIntegritySpec.groovy

**Comment by paplorinc** [comment]
Ah, @JLLeitschuh, you mean the failing test, pushed :)

**Pull request merged**